File: Tema8/EntregaCRUDProductos/app/acciones.php
<?php
include_once "Producto.php";

function accionBorrar ($num_product){    
    $db = AccesoDatos::getModelo();
    $tuser = $db->borrarProducto($num_product);
}

function accionTerminar(){
    AccesoDatos::closeModelo();
    session_destroy();
}
 
function accionAlta(){
    $product = new Producto();
    $product->PRODUCTO_NO  = "";
    $product->DESCRIPCION   = "";
    $product->PRECIO_ACTUAL   = "";
    $product->STOCK_DISPONIBLE = "";
    $orden= "Nuevo";
    include_once "layout/formulario.php";
}

function accionDetalles($num_product){
    $db = AccesoDatos::getModelo();
    $product = $db->getProducto($num_product);
    $orden = "Detalles";
    include_once "layout/formulario.php";
}


function accionModificar($num_product){
    $db = AccesoDatos::getModelo();
    $product = $db->getProducto($num_product);
    $orden="Modificar";
    include_once "layout/formulario.php";
}

function accionPostAlta(){
    limpiarArrayEntrada($_POST); //Evito la posible inyección de código
    $product = new Producto();
    $product->PRODUCTO_NO  = $_POST['numero'];
    $product->DESCRIPCION   = $_POST['descripcion'];
    $product->PRECIO_ACTUAL   = $_POST['precio'];
    $product->STOCK_DISPONIBLE = $_POST['stock'];
    $db = AccesoDatos::getModelo();
    $db->addProducto($product);
    
}

function accionPostModificar(){
    limpiarArrayEntrada($_POST); //Evito la posible inyección de código
    $product = new Producto();
    $product->PRODUCTO_NO  = $_POST['numero'];
    $product->DESCRIPCION   = $_POST['descripcion'];
    $product->PRECIO_ACTUAL   = $_POST['precio'];
    $product->STOCK_DISPONIBLE = $_POST['stock'];
    $db = AccesoDatos::getModelo();
    $db->modProducto($product);
    
}

File: ProductosMVC/src/main/java/controladores/Acciones.java
package controladores;

import modelo.AccesoDatos;
import modelo.Producto;

import java.io.IOException;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

public class Acciones {

	HttpServletRequest request;
	HttpServletResponse response;

	Acciones(HttpServletRequest request, HttpServletResponse response) {
		this.request = request;
		this.response = response;
	}

	void accionAlta() throws ServletException, IOException {
		Producto prod = new Producto();
		request.setAttribute("orden", "Nuevo");
		request.setAttribute("prod", prod);
		request.getRequestDispatcher("/WEB-INF/layout/formulario.jsp").forward(request, response);
	}

	void accionBorrar(int id) {
		AccesoDatos db = AccesoDatos.initModelo();
		db.borrarProducto(id);
	}

	void accionModificar(int id) throws ServletException, IOException {
		AccesoDatos db = AccesoDatos.initModelo();
		Producto prod = db.getProducto(id);
		request.setAttribute("orden", "Modificar");
		request.setAttribute("prod", prod);
		request.getRequestDispatcher("/WEB-INF/layout/formulario.jsp").forward(request, response);

	}

	void accionDetalles(int id) throws ServletException, IOException {
		AccesoDatos db = AccesoDatos.initModelo();
		Producto prod = db.getProducto(id);
		request.setAttribute("orden", "Detalles");
		request.setAttribute("prod", prod);
		request.getRequestDispatcher("/WEB-INF/layout/formulario.jsp").forward(request, response);
	}

	void accionTerminar() {
		System.out.println("->>> accionTerminar   \n");
	}

	void accionPostAlta() {
		// Habría que controlar los datos de recibido <<<<<<<

		Producto prod = new Producto();
		prod.setProducto_no(Integer.parseInt(request.getParameter("producto_no")));
		prod.setDescripcion(request.getParameter("descripcion"));
		//Controlamos que no ocurra un error NumberFormatException al intentar convertir a Double con una coma		
		prod.setPrecio_actual(Double.parseDouble((request.getParameter("precio_actual").replace(",", "."))));
		prod.setStock_disponible((Integer.parseInt(request.getParameter("stock_disponible"))));
		AccesoDatos db = AccesoDatos.initModelo();
		db.addProducto(prod);
	}

	void accionPostModificar() {
		// Habría que controlar los datos de recibido <<<<<<<
		Producto prod = new Producto();
		prod.setProducto_no(Integer.parseInt(request.getParameter("producto_no")));
		prod.setDescripcion(request.getParameter("descripcion"));		
		//Controlamos que no ocurra un error NumberFormatException al intentar convertir a Double con una coma	
		prod.setPrecio_actual(Double.parseDouble((request.getParameter("precio_actual").replace(",", "."))));
		prod.setStock_disponible((Integer.parseInt(request.getParameter("stock_disponible"))));
		AccesoDatos db = AccesoDatos.initModelo();
		db.modProducto(prod);

	}
}
